### PR TITLE
fix: ReadObject byte range

### PR DESCRIPTION
**Pull request by laramiel**
fix: gRPC ReadObject byte range handling:
* Byte range reads now return the requested byte range.
* Zero length values now yield at least one response.

### CODE REVIEW

**Review by coryan — CHANGES_REQUESTED** [review]

File: testbench/grpc_server.py [hunk 1]
```diff
@@ -485,10 +485,15 @@ def ReadObject(self, request, context):
             preconditions=testbench.common.make_grpc_preconditions(request),
         )
         size = storage_pb2.ServiceConstants.Values.MAX_READ_CHUNK_BYTES
+        start = request.read_offset
```
[the remark this answers is not in the document]
Comment (laramiel) [reply testbench/grpc_server.py:489]: Hm, the tests aren't working for me either. Let me see if I can figure it out.

.
======================================================================
FAIL: test_read_object_byte_range (__main__.TestGrpc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/storage-testbench/tests/test_grpc_server.py", line 1000, in test_read_object_byte_range
    self.assertEqual(1, len(chunks))
AssertionError: 1 != 3

======================================================================
FAIL: test_read_zero_size_object (__main__.TestGrpc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/storage-testbench/tests/test_grpc_server.py", line 1018, in test_read_zero_size_object
    self.assertEqual(1, len(chunks))
AssertionError: 1 != 0

----------------------------------------------------------------------
Ran 63 tests in 1.469s

[the remark this answers is not in the document]
Comment (laramiel) [reply testbench/grpc_server.py:489]: Ok, I figured out the problem. I had an old version installed in my SitePackages, causing my tests to fail. Updated to fix the failure you saw.

**Review by coryan — CHANGES_REQUESTED** [review]
This looks great, thanks.  Please make the formatting robots happy so we can merge this:

https://github.com/googleapis/storage-testbench/actions/runs/4420456037/jobs/7750174123?pr=476

**Comment by laramiel** [comment]
> This looks great, thanks. Please make the formatting robots happy so we can merge this:
> 
> https://github.com/googleapis/storage-testbench/actions/runs/4420456037/jobs/7750174123?pr=476

Done. I think.

**Review by coryan — APPROVED** [review]
Thanks for the PR!

**Pull request merged**